### PR TITLE
75636 replace transform validation and tests

### PR DESCRIPTION
**Pull request by dawidkeshop**

### CODE REVIEW

File: src/CaptainHook.EventHandlerActor/Validation/WebhookRequestRuleForRouteAndReplaceValidator.cs [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+            return routes?.Select(x => x.Selector).Distinct().Count() == routes?.Count;
+        }
+
+        private class SourceParserLocationValidator : AbstractValidator<SourceParserLocation>
+        {
+            public SourceParserLocationValidator()
+            {
+                RuleFor(x => x.Location).Equal(Location.Body);
+                RuleFor(x => x.Path).Null();
+                RuleFor(x => x.Replace).Must(kvp => kvp?.ContainsKey("selector") == true);
```
Comment (mszymeshop) [inline src/CaptainHook.EventHandlerActor/Validation/WebhookRequestRuleForRouteAndReplaceValidator.cs:30]: there is a const you can reuse

File: src/CaptainHook.EventHandlerActor/Validation/WebhookRequestRuleForRouteAndReplaceValidator.cs [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+        private bool HaveUniqueSelectors(List<WebhookConfigRoute> routes)
+        {
+            return routes?.Select(x => x.Selector).Distinct().Count() == routes?.Count;
+        }
+
+        private class SourceParserLocationValidator : AbstractValidator<SourceParserLocation>
+        {
+            public SourceParserLocationValidator()
+            {
+                RuleFor(x => x.Location).Equal(Location.Body);
```
Comment (mszymeshop) [inline src/CaptainHook.EventHandlerActor/Validation/WebhookRequestRuleForRouteAndReplaceValidator.cs:28]: I don't think we care about the location value for Source

File: src/Tests/CaptainHook.Tests/TestsInfrastructure/GetInvalidEnumValuesAttribute.cs [hunk 3]
```diff
@@ -0,0 +1,48 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using Xunit.Sdk;
+
+namespace CaptainHook.Tests.TestsInfrastructure
+{
+    public class GetInvalidEnumValuesAttribute : DataAttribute
```
Comment (mszymeshop) [inline src/Tests/CaptainHook.Tests/TestsInfrastructure/GetInvalidEnumValuesAttribute.cs:9]: I don't seem to see usages of that

File: src/CaptainHook.EventHandlerActor/Validation/WebhookRequestRuleForRouteAndReplaceValidator.cs (hunk 1 above)
Comment (dawidkeshop) [reply src/CaptainHook.EventHandlerActor/Validation/WebhookRequestRuleForRouteAndReplaceValidator.cs:30]: Ok

File: src/Tests/CaptainHook.Tests/TestsInfrastructure/GetInvalidEnumValuesAttribute.cs (hunk 3 above)
Comment (dawidkeshop) [reply src/Tests/CaptainHook.Tests/TestsInfrastructure/GetInvalidEnumValuesAttribute.cs:9]: It's used in three test theories.

File: src/CaptainHook.EventHandlerActor/Validation/WebhookRequestRuleForRouteAndReplaceValidator.cs (hunk 2 above)
Comment (dawidkeshop) [reply src/CaptainHook.EventHandlerActor/Validation/WebhookRequestRuleForRouteAndReplaceValidator.cs:28]: Ok, will remove this rule and according tests.

File: src/Tests/CaptainHook.Tests/TestsInfrastructure/GetInvalidEnumValuesAttribute.cs (hunk 3 above)
Comment (mszymeshop) [reply src/Tests/CaptainHook.Tests/TestsInfrastructure/GetInvalidEnumValuesAttribute.cs:9]: I see now, clever :)

File: src/CaptainHook.EventHandlerActor/Validation/WebhookConfigForRouteAndReplaceValidator.cs [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+        }
+
+        private static bool NotContainRulesWithRuleActionRoute(List<WebhookRequestRule> rules)
+        {
+            return rules.All(rule => rule.Destination?.RuleAction != RuleAction.Route);
+        }
+
+        private static bool NotContainRulesWithRuleActionAddAndLocationUri(List<WebhookRequestRule> rules)
+        {
+            return rules.All(rule => !(rule.Destination?.RuleAction == RuleAction.Add && rule.Destination?.Location == Location.Uri));
```
Comment (mszymeshop) [inline src/CaptainHook.EventHandlerActor/Validation/WebhookConfigForRouteAndReplaceValidator.cs:38]: I think `rule.Destination?.RuleAction == RuleAction.Add` is a bit unknown. We can only refer to Uri location at the moment. We can leave it though and see where it leads us :)

**Pull request merged**